recipes/boost/all/conanfile.py

```diff
@@ -120,6 +122,10 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def configure(self):
+        if self.options.icu:
```

Comment:
```suggestion
        if self.options.icu and self.settings.get_safe("cppstd"):
```

cppstd is not configured by default and we don't distribute packages with cppstd. GCC 4.9 supports C++11 (most of features) but doesn't run C++11 by default.

Reply:
¿?

Reply:
One line below you have: `tools.check_min_cppstd(self, "11")`

Which means, if cppstd is not declared, then the default cpp standard support by the compiler will be used. However, GCC 4.9 supports C++11, but is not the default one. Of course, the project (Boost) needs to configure c++11 internally.

Reply:
you still may build boost for C++03 (without ICU).
that's ICU who should set these constrains. thus, ICU is he one who should use the check.

Reply:
Totally agree, this check is for ICU, not here https://github.com/conan-io/conan-center-index/pull/1274#issuecomment-609633928, please review with a 🔴 so this is not merged accidentally, and we will talk on Wednesday about cppstd for sure

Reply:
Good point!

Reply:
I'm resolving this conversation, with all we have already talked about `cppstd` I think that the underlying problem is known by everyone of us. Now we need a solution. And I fixed the PR to avoid everything that doesn't belong to this library.

Please, review from scratch. Thanks!